pkg/discovery/worker/k8s_discovery_worker.go

```diff
@@ -209,7 +206,7 @@ func (worker *k8sDiscoveryWorker) executeTask(currTask *task.Task) *task.TaskRes
 					glog.V(3).Infof("A %s monitoring worker from discovery worker %v is invoked for task %s.",
 						w.GetMonitoringSource(), worker.id, resourceMonitorTask)
 					// Assign task to monitoring worker.
-					monitoringSink := w.Do(stopCh)
+					monitoringSink := w.Do()
```

Comment:
Shall we also return possible `error` for the `Do()` function? In the case of error, we don't need to merge sink later on.

Reply:
Added. PTAL